src/app/features/filter-bar/reducers/filter.reducer.ts

```diff
@@ -19,33 +9,40 @@ export class FilteringState {
   searchValue: string;
   priceValue: PriceFilter;
   categoryValue: string[];
+  countryValue: string[];
   cityValue: string[];
 }
 
 export let initialState: FilteringState = {
   searchValue: '',
   priceValue: { from: 0, to: Number.MAX_SAFE_INTEGER },
   categoryValue: undefined,
+  countryValue: undefined,
   cityValue: undefined,
 };
 
 export const filteringReducer = createReducer(
   initialState,
   on(filterBySearch, (state, action) => ({
     ...state,
-    searchValue: action.value,
+    searchValue: action.value
   })),
   on(filterByPrice, (state, action) => ({
     ...state,
-    priceValue: { from: action.from, to: action.to },
+    priceValue: { from: action.from, to: action.to }
   })),
   on(filterByCategory, (state, action) => ({
     ...state,
-    categoryValue: getNewFilteredValues(action.value, state.categoryValue),
+    categoryValue: getNewFilteredValues(action.value, state.categoryValue)
+  })),
+  on(filterByCountry, (state, action) => ({
+    ...state,
+    countryValue: getNewFilteredValues(action.value, state.countryValue),
+    cityValue: undefined
```

Comment:
Why are we doing it?
And if there is a reason, why not doing the same on the filterByCity?

Reply:
The reason is because when the user change the country filter we can't make sure that he cityValue containing only cities from the countries which selected, so the easiest way is just to initialize this value

Reply:
Sound about right